src/ngraph/op/experimental/transpose.cpp

```diff
@@ -0,0 +1,59 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <iostream>
+
+#include "ngraph/op/experimental/transpose.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+op::Transpose::Transpose(const shared_ptr<Node>& arg, const shared_ptr<Node>& input_order)
+    : Op("Transpose", check_single_output_args({arg, input_order}))
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::Transpose::validate_and_infer_types()
+{
+    NODE_VALIDATION_CHECK(this,
+                          get_output_element_type(0).compatible(element::i64),
```

Comment:
Should this be `get_input_element_type(1)`?

Reply:
Yes! Good catch. Please request changes.